src/ansys/fluent/core/file_session.py

```diff
@@ -724,3 +708,24 @@ def read_case(self, case_file_name):
     def read_data(self, data_file_name):
         """Read Data file."""
         self._data_file = DataFile(data_file_name, case_file_handle=self._case_file)
+
+
+class TransactionError(RuntimeError):
+    """Custom transaction errors."""
+
+    def __init__(self, error):
+        super().__init__(error)
+
+    @classmethod
+    def SURFACE_NAMES_IDs_NOT_PROVIDED(cls):
```

Comment:
At the consumer level, these exception objects are still of type TransactionError and the way to identify a SURFACE_NAMES_IDs_NOT_PROVIDED exception is to check for the error message. Do we want to refine the exception hierarchy further by deriving a SurfaceNotSpecified from TransactionError?

All the custom exception classes can be moved to a separate file, so user can find them in one place.

Reply:
Discussed with @mkundu1, we will refine the exception hierarchy. Thank you.

Reply:
@hpohekar you could also look into refining exception hierarchy in the units repo if it interests you - I believe that what we are doing there is similar to what we currently see here.

Reply:
Yes, after completing this task I will look into it.

Reply:
@hpohekar @seanpearsonuk Another usage pattern could be:
```
try:
    <pyfluent code>
except TransactionError as ex:
    if ex.code == TransactionError.ErrorCode.SURFACE_NAMES_IDS_NOT_PROVIDED:
        <handle SURFACE_NAMES_IDS_NOT_PROVIDED exception>
    elif ex.code == TransactionError.ErrorCode.INVALID_FIELD_NAME_PREFIX:
        <handle INVALID_FIELD_NAME_PREFIX exception>
    else:
        <handle other TransactionError>
```
`ErrorCode` is an enum. This way we won't have too many custom exception classes, but users will be able to identify the fine-grained errors.

Reply:
I personally would vote for using and catching specific exception types if possible, like `SurfaceNotSpecifiedError` as you said earlier, otherwise that structure you just shared is going back to being very close to just parsing the Exception error message as in this comment and my comment after that: https://github.com/ansys/pyfluent/issues/2098#issuecomment-1750759704

Which doesn't require any custom exception classes.

Reply:
They do have this:
![image](https://github.com/ansys/pyfluent/assets/132297401/a1e794af-41fb-4c65-bd6b-8be8085a3e8a)
Which is closer to what I believe we should be aiming for if we are going to add more custom exception classes:

Reply:
@raph-luc There is a lack of official documentation here. [This](https://github.com/ansys/pyfluent-reader/pull/14#pullrequestreview-1688178548) is the general usage pattern for handling exceptions from a Python client.

Right, internally they have implemented a full custom class hierarchy, but that implementation is private.

Reply:
I see, thanks for the explanation. As they have those status codes which are used repeatedly and relevant everywhere in their modules (see https://grpc.github.io/grpc/core/md_doc_statuscodes.html), it makes sense that they would include the status codes as part of their custom exception classes.

Reply:
@mkundu1 This looks good to me. Thank you.

![image](https://github.com/ansys/pyfluent/assets/106588300/4b59f669-ac15-428b-85ce-405d350b0318)